src/core/utils/misc.cpp

```diff
@@ -526,6 +532,64 @@ QString Utils::Misc::parseHtmlLinks(const QString &raw_text)
     return result;
 }
 
+#ifndef DISABLE_GUI
+// Open the given path with an appropriate application
+void Utils::Misc::openPath(const QString& absolutePath)
+{
+    const QString path = Utils::Fs::fromNativePath(absolutePath);
+    // Hack to access samba shares with QDesktopServices::openUrl
+    if (path.startsWith("//"))
+        QDesktopServices::openUrl(Utils::Fs::toNativePath("file:" + path));
+    else
+        QDesktopServices::openUrl(QUrl::fromLocalFile(path));
+}
+
+// Open the parent directory of the given path with a file manager and select
+// (if possible) the item at the given path
+void Utils::Misc::openFolderSelect(const QString& absolutePath)
+{
+    const QString path = Utils::Fs::fromNativePath(absolutePath);
+#ifdef Q_OS_WIN
+    if (QFileInfo(path).exists()) {
+        // Syntax is: explorer /select, "C:\Folder1\Folder2\file_to_select"
+        // Dir separators MUST be win-style slashes
+        QProcess::startDetached("explorer.exe", QStringList() << "/select," << Utils::Fs::toNativePath(absolutePath));
+    }
+    else {
+        // If the item to select doesn't exist, try to open its parent
+        openPath(path.left(path.lastIndexOf("/")));
+    }
+#elif defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
+    if (QFileInfo(path).exists()) {
+        QProcess proc;
+        QString output;
+        proc.start("xdg-mime", QStringList() << "query" << "default" << "inode/directory");
+        proc.waitForFinished();
+        output = proc.readLine().simplified();
+        if (output == "dolphin.desktop")
```

Comment:
@sledgehammer999 I found a bug. `dolphin.desktop` is for KDE4. Replace the line with this to work in KDE4 and KDE5.
```
if (output == "dolphin.desktop" || output == "org.kde.dolphin.desktop")
```